File: StreamProviderWS/StreamProviderWS/Models/Common/MovieRoom.cs
using System.Collections.Generic;

namespace StreamProviderWS.Models.Common
{
    public class MovieRoom : Base
    {
        public Movie Movie { get; set; }

        public List<UserRoom> UsersInRoom { get; set; }

        public string Stream { get; set; }

        public double TimeWatched { get; set; }
    }
}

File: VideoStreamerUI/src/app/entities/movieRoom.ts
import { Movie } from './movie';
import { UserInRoom } from './userInRoom';

export class MovieRoom {
    Id: string;
    Movie: Movie;

    // list of users that have joined this room at least once
    UsersInRoom: UserInRoom[];

    // video stream
    Stream: string;
    TimeWatched: number;

    constructor(movie: Movie, usersInRoom: UserInRoom[], stream: string, timeWatched: number) {
        this.Movie = movie;
        this.UsersInRoom = usersInRoom;
        this.Stream = stream;
        this.TimeWatched = timeWatched;
    }
}
